proto/buoyant-cloud-api.proto

```diff
@@ -92,11 +92,69 @@ message Event {
   Workload owner = 3;
 }
 
+//
+// Linkerd Messages
+//
+
+// This message represents PEM encoded certificate
+// data. It might be a single certificate, a pool of
+// roots or a chain of certificates.
+message CertData {
+  bytes raw = 1;
+}
+
+// Represents the certificates that the control plane
+// has been configured with.
+message ControlPlaneCerts {
+  // This is the identity issuer certificate chain
+  CertData issuer_crt_chain = 1;
+  // The roots that the control plane has been configured with
+  CertData roots = 2;
+}
+
+// A message that carries one or more pod certificates
+message ProxyCerts {
+  repeated ProxyCert certs = 1;
+
+  message ProxyCert {
+    // The peer chain which include the issuer
+    // cert as well as the leaf certificate
+    CertData peer_crt_chain = 1;
+    // The roots this proxy has been configured with
+    CertData roots = 2;
+    string pod_name = 3;
+    string namespace = 4;
+    string workload_name = 5;
+  }
+}
+
+message WebhookCerts {
+  // The certificate chain for this webhook cert
+  CertData crt_chain = 1;
+  string component_name = 2;
+}
+
+message CertificateInfo {
+  oneof info {
+    ControlPlaneCerts control_plane = 1;
+    ProxyCerts proxy = 2;
+    WebhookCerts webhook = 3;
+  }
+}
+
+message LinkerdMessage {
+  Auth auth = 1;
+  oneof message {
+    CertificateInfo crt_info = 2;
+  }
+}
+
 //
 // API
 //
 
 service Api {
   rpc WorkloadStream(stream WorkloadMessage) returns (Empty) {}
   rpc AddEvent(Event) returns (Empty) {}
+  rpc LinkerdInfo(LinkerdMessage) returns (Empty) {}
```

Comment:
I don't have a lot of context, but would it make sense for each of these message types (i.e. each type of certificate info) to have its own api endpoint, rather than vary them within the body? It probably doesn't really matter, but if we considered a router/proxy in between, it would actually have to know about the message body to make decisions. If they're split into different API endpoints, they can have separate success rates, policies, etc....